### DISCUSSION TITLE
synthetic turbulent inflow

### DISCUSSION THREAD

**Discussion by indra098124** [Exawind/amr-wind#819] [Q&A] [answered]
Hi, I am trying to set a synthetic turbulent inflow for an AMR-wind simulation.  I see that synthetic turbulent inflow needs a file ("https://exawind.github.io/amr-wind/api_docs/group__synth__turb.html#details"), how do I get this file. I want to prescribe a known mean velocity, temperature and humidity profile at the inlet with synthetic turbulence imposed. 
Thank you

**Comment by marchdf** [comment]
I just received this email (extracted only the relevant part): :For my research, I am mostly working with amr-wind and am planning to set up several simulations using synthetic turbulence. Unfortunately, I don’t know how. I tried looking through the documentation but couldn’t find a simple solution. "

It seems like we should improve this documentation. Does anyone here have better instructions?

And apologies for never noticing this open question, almost a year old. If you solved the problem, @indra098124, great, feel free to share. If not, sorry this never got resolved.

**Comment by AlexGuntner** [comment]
Hi @indra098124,

have you ever managed to set up a case using synthetic turbulent inflow? I am looking for some examples to understand what file format is required, but I cannot find any in the AMR-Wind repository.

Best

**Comment by marchdf** [comment]
@gantech checked in this capability 4 years ago but we don't have a regtest. @gantech would you mind sharing a sample netcdf file? Or do you know anyone else who's used this capability?

**Reply by AlexGuntner** [reply]
Dear @marchdf,

thank you for your reply.

For my work, I would also like to run a few laminar inflow cases using LES with the Smagorinsky model. To do this, I attempted to simulate the uniform ALM case ("test/test_files/act_uniform_alm") by changing "turbulence.model" from Laminar to Smagorinsky. However, I haven’t been able to run a laminar case with an LES model enabled without getting startup errors. Is this feature supported? 

Additionally, the documentation is somewhat unclear on what "Laminar" actually means beyond stating that it is "effectively no turbulence model." Could you clarify this or direct me to relevant documentation?

Best

**Reply by marchdf** [reply]
"Laminar" (in the code) means there is no eddy viscosity. The only relevant viscosity is the molecular one. 

I am a bit puzzled about wanting to run a laminar flow with an LES model. But regardless, you can activate whatever LES model you want with whatever flow field you want.

**Comment by AlexGuntner** [comment]
Dear @marchdf,

Thank you for the clarification. My primary reason for running LES with a laminar inflow is for validation, as it allows for easier comparison with existing data. Additionally, since I aim to investigate how platform motions influence wind turbine wake dynamics, using a laminar inflow helps highlight the key wake characteristics more clearly.

I will try rerunning the simulations.

Best

**Comment by sTamaroTum** [comment]
Hi all,

I am also trying to do this. So far I managed to create a .nc file and it is read correctly. The simulation starts without crashing, but the effect of my SyntheticTurb is none. Even if I put super high epsilon values nothing happens, so I guess that I am missing something in the definition of my turbulence box. 

Best
Simone

**Reply by marchdf** [reply, marked as answer]
yes that could be. Did you check out this example: https://github.com/lawrenceccheung/SyntheticTurbulenceTest/tree/master/amrwind_run01

**Reply by sTamaroTum** [reply]
Yes, thank you so much. Basically i had omitted:

ICNS.source_terms = SynthTurbForcing

Now I see the velocity fields perturbed. Many thanks for the prompt help!
